npm/vite-dev-server/src/plugins/cypress.ts

```diff
@@ -105,6 +105,23 @@ export const Cypress = (
         return res.end(transformedIndexHtml)
       })
     },
+    transform (code, id, options?) {
+      try {
+        if (/\.js$/.test(id) && !/\/\/# sourceMappingURL=/i.test(code)) {
+          // The Vite dev server serves esnext, meaning many JS files aren't transpiled thus no sourcemaps are sent along. However,
+          // a sourcemap is generated by Vite which is in effect an "identity" sourcemap mapping 1-to-1 to the output file. We can
+          // grab this and pass it along as a sourcemap we want Vite to embed into the output, giving Cypress a sourcemap to use
+          // for codeFrame lookups.
```

Comment:
```suggestion
          // The Vite dev server does not perform transpilation. Source code is served as it is written, thus no sourcemaps are generated by default. However,
          // Vite uses Rollup internally to support code transformation, which is in effect an "identity" sourcemap mapping 1-to-1 to the output file. Vite gives us access to the underyling Rollup API, including `getCombinedSourcemap`. 
          // We can grab this and pass it along as a sourcemap we want Vite to embed into the output, giving Cypress a sourcemap to use
          // for codeFrame lookups.
          // @see https://rollupjs.org/guide/en/#thisgetcombinedsourcemap
```

Reply:
Not suggesting using this as-is, but I think we might want to be quite descriptive about what's actually happening here, it's quite complex and unless you know exactly how Vite works, it would be quite difficult to make the connection to Vite -> Rollup -> `getCombinedSourcemap`.

Reply:
@lmiller1990 I've updated the comment a bit to provide more context, let me know if you would like to see additional changes

Reply:
Looks good!